src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -1104,10 +1238,15 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
         }
 
         protected virtual TagBuilder GenerateLink(
-            [NotNull] string linkText,
-            [NotNull] string url,
+            string linkText,
+            string url,
```

Comment:
Ok. We'll need to update a couple of functional tests in that case. Essentially if you have an anchor tag helper with incomplete information such as: `<a asp-controller="something">Some value</a>` - we'll throw a Arg Null Exception.

cc @NTaylorMullen